sw/cheri/tests/hyperram_tests.hh

```diff
@@ -217,6 +217,8 @@ void write_prog(Capability<volatile uint32_t> hyperram_area, uint32_t addr) {
   hyperram_area[addr + 2] = 0x00552023;
   hyperram_area[addr + 3] = 0x00000517;
   hyperram_area[addr + 4] = 0x8082;
+
+  asm volatile ("fence.i");
```

Comment:
Technically not needed because `hyperram_area` is volatile, but I think it's still worth adding to capture that memory op reordering must not happen for this instruction:

```suggestion
  asm volatile ("fence.i" ::: "memory");
```